### ISSUE TITLE
Difference in validation accuracy

### ISSUE THREAD

**Issue by gehao513** [keras-team/keras#7759]
Hi guys, I am new to keras and find a interesting question.
X_data is my dataset and we can find the following two expressions of X_sample should be the same.

Expression 1:

sample = random.sample(range(len(Y)), int(floor(1.00*len(Y))))
print(sample)
X_sample = X_data[sample, :]                              
Y_sample = Y[sample]

Expression 2:
X_sample = X_data
Y_sample = Y

I build a model and fix the data:
model.fit(X_sample, y_sample, validation_split=0.33, epochs=100, batch_size=200) 

The strange thing is, the training accuracy is close to one, but for the first expression, the validation accuracy is close to 1 as well, while for the second expression, the validation accuracy is about 0.36.

Can anyone explain this for me? Thanks a lot!
